### PR TITLE
[bugfix] show_fix should set num_workers to 0

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -148,3 +148,4 @@ imdb
 xsum
 coco128
 wmt_en_ro
+kinetics
```
Comment (ethanwharris) [inline .gitignore:151]: Is this from something else?

File: flash/data/data_module.py [hunk 2]
```diff
@@ -138,8 +138,12 @@ def data_fetcher(self, data_fetcher: BaseDataFetcher) -> None:
 
     def _reset_iterator(self, stage: RunningStage) -> Iterable[Any]:
         iter_name = f"_{stage}_iter"
+        # the num_workers has to be set to 0 to work properly.
```
Comment (ethanwharris) [inline flash/data/data_module.py:141]:
```suggestion
        # num_workers has to be set to 0 to work properly
```

File: tests/data/test_callbacks.py [hunk 3]
```diff
@@ -181,3 +181,15 @@ def extract_data(data):
         assert dm.data_fetcher.show_collate_called
         assert dm.data_fetcher.per_batch_transform_called
         dm.data_fetcher.check_reset()
+
+
+def test_data_loaders_num_workers_to_0(tmpdir):
+    """
+    The num_workers should be set to `0` internally for visualization and not for training.
```
Comment (ethanwharris) [inline tests/data/test_callbacks.py:188]:
```suggestion
    num_workers should be set to `0` internally for visualization and not for training.
```

File: flash/data/data_module.py [hunk 4]
```diff
@@ -191,7 +195,8 @@ def get_dataset_attribute(dataset: torch.utils.data.Dataset, attr_name: str, def
     def set_dataset_attribute(dataset: torch.utils.data.Dataset, attr_name: str, value: Any) -> None:
         if isinstance(dataset, Subset):
             dataset = dataset.dataset
-        setattr(dataset, attr_name, value)
+        if isinstance(dataset, (Dataset, IterableDataset)):
+            setattr(dataset, attr_name, value)
```
Comment (ethanwharris) [inline flash/data/data_module.py:199]: Should there be a warning or something here if it is not in [Subset, Dataset, Iterable]?

**Review by ethanwharris — APPROVED** [review]
LGTM :smiley: small nits

File: .gitignore (hunk 1 above)
Comment (tchaton) [reply .gitignore:151]: It is the video dataset. I have them installed and in s3. Just preventing them to be added.

File: flash/data/data_module.py (hunk 2 above)
Comment (tchaton) [reply flash/data/data_module.py:141]: Done !

File: flash/data/data_module.py (hunk 4 above)
Comment (tchaton) [reply flash/data/data_module.py:199]: I don't think so, it is really for internal tracking.

File: tests/data/test_callbacks.py [hunk 5]
```diff
@@ -181,3 +181,15 @@ def extract_data(data):
...
+    """
+
+    datamodule = DataModule(train_dataset=range(10), num_workers=3)
+    iterator = datamodule._reset_iterator(RunningStage.TRAINING)
+    assert isinstance(iterator, torch.utils.data.dataloader._SingleProcessDataLoaderIter)
+    iterator = iter(datamodule.train_dataloader())
+    assert isinstance(iterator, torch.utils.data.dataloader._MultiProcessingDataLoaderIter)
```
Comment (edgarriba) [inline tests/data/test_callbacks.py:195]: I'd add a small assert checking that internally the number of workers has been been kept as expected

Comment (tchaton) [reply tests/data/test_callbacks.py:195]: That's what `_MultiProcessingDataLoaderIter ` does.

**Pull request merged**